junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -93,7 +93,7 @@ void notifyFinished() {
         switch (state) {
         case INIT:
             notifyPrepared();
-            break;
+//            break;
```

Comment:
Connection to the driver is done in the additional thread. So prepare also is on that thread. The main thread starts the second thread then waits for prepared notification. Only after prepared the actual test is started. Check SpecificationStatement.evaluate().